meta: Remove black GH action

sl0thentr0py opened the pull request:
Since we moved to `pre-commit` in #1315 anyway, we don't need this flaky black stuff to run on push.
The `lint` step checks with `black --check` either way.

antonpirker commented:
I am also for removing it.

One thing we might keep in mind: people submitting PRs maybe do not have pre-commit installed so can submit horribly formatted code.

sl0thentr0py commented:
I wanted to bring this up in the pre-commit PR but anyway, let's do it now.
We already have `make format` [here](https://github.com/getsentry/sentry-python/blob/384cf8741c7d493dd84b6be939d1b068962adf6f/1#L27-L30), now that we added pre-commit, that's redundant.
I initially thought why add a new tool when we can just run `make format` in a simple git pre-commit hook?
But anyway, now that we added it, we can just change the `Makefile` and make users setup `pre-commit` in `CONTRIBUTING.md`.
